# Analysis using Stanford website data

** Outline: what to be done** 
1. original txt files from database: PR.txt and RT.txt

2. 4 fasta files  (plus description of how you aligned them)

3. one file with frequencies for 984 sites (plus R script to prepare this file)

4. plot of bacheler vs stanford data, plus R^2 values & p values (take out drug resistance sites, and first 13 codons of protease) 

5. maybe also plots, R^2 values for only syn sites, only non-syn sites. 

6. short text for material and methods section of the paper to describe everything





## Part 1 :  Download and preprocessing

### Protease download and fasta file generation 
The dataset that you can download from Stanford (http://hivdb.stanford.edu/pages/geno-rx-datasets.html) is called. PR.txt and RT.txt (see below)

We will process this file to remove non-B subtypes, empty sequences, treated sequences, multiple sequences per patient and sequences containing bad characters. 


#### retrieve sequences from PR.xt and clean up 
```{r}
data<-read.csv('PR.txt',sep=',')
# only subtype B
dataB<-subset(data,data$Subtype=='B')
# remove empty sequences 
dataBn<-subset(dataB,dataB$NASeq!="")
# only naive sequences 
dataBnaive<-subset(dataBn,dataBn$PIList=='None')
dataBtreated<-subset(dataBn,dataBn$PIList!='None')
# only 1 sequence per patient 
dataBtreated_unique<-dataBtreated[!duplicated(dataBtreated$PtID),]
dataBnaive_unique<-dataBnaive[!duplicated(dataBnaive$PtID),]
# remove sequences with bad characters
dataBnaive_unique2<-dataBnaive_unique[-grep('~',dataBnaive_unique$NASeq),]
dataBtreated_unique2<-dataBtreated_unique#[-grep('~',dataBtreated_unique$NASeq),]

# show evolution in size
nrow(data);nrow(dataB);nrow(dataBn);nrow(dataBnaive);nrow(dataBnaive_unique);nrow(dataBtreated);nrow(dataBtreated_unique);nrow(dataBnaive_unique2);nrow(dataBtreated_unique2)

# combine  identifier en sequence
##export naive sequences but still not aligned
dataBnaive_unique3<-cbind(as.character(dataBnaive_unique2$PtID),as.character(dataBnaive_unique2$NASeq))
write.table(as.data.frame(dataBnaive_unique3),file='subtypeB-pr_naive.csv',row.names=F,quote=F,sep=',')
##export treated sequences but still unaligned
dataBtreated_unique3<-cbind(as.character(dataBtreated_unique2$PtID),as.character(dataBtreated_unique2$NASeq))
write.table(as.data.frame(dataBtreated_unique3),file='subtypeB-pr_treated.csv',row.names=F,quote=F,sep=',')
```


#### generate a aligned fasta file from the sequences 
Several steps to convert the csv with sequences into a fasta file

** to fasta format **
vi subtypeB-pr_naive.csv # add >
vi subtypeB-pr_treated.csv
tr ',' '\n' < subtypeB-pr_naive.csv > subtypeB-pr_naive.fasta
tr ',' '\n' < subtypeB-pr_treated.csv > subtypeB-pr_treated.fasta

** transfer to server **
scp *fasta cai:/home/ktheys0
grep '\.' subtypeB-rt_naive.fasta  |wc

** Align sequences using C++ library  **
This tool for pairwise alignment against HXB2 is available at http://regatools.med.kuleuven.be/sequencetool/sequencetool.wt

SequenceTool PR subtypeB-pr_naive.fasta --exportKind GlobalAlignment --exportAlphabet Nucleotides --exportWithInsertions no  > subtypeB-pr_naive_aligned.fasta
SequenceTool PR subtypeB-pr_treated.fasta --exportKind GlobalAlignment --exportAlphabet Nucleotides --exportWithInsertions no  > subtypeB-pr_treated_aligned.fasta

** remove HXB2 from alignement ** 
vi subtypeB-pr_naive_aligned.fasta
vi subtypeB-pr_treated_aligned.fasta

** Convert nucleotide sequences to amino acids:   **
SequenceTool PR subtypeB-pr_naive_aligned.fasta --exportKind PositionTable  > subtypeB-pr_naive_mut.csv
SequenceTool PR subtypeB-pr_treated_aligned.fasta --exportKind PositionTable   > subtypeB-pr_treated_mut.csv

** get from server  ** 
scp cai:/home/ktheys0/subtypeB-pr* .
scp cai:/home/ktheys0/subtypeB-pr*csv .


### Reverse transcriptase  download and fasta file generation 
Similar steps as for the PR.txt file
#### retrieve sequences from RT.txt
```{r}
#same steps as for PR
data<-read.csv('RT.txt',sep='\t')
dataB<-subset(data,data$Subtype=='B')
dataBn<-subset(dataB,dataB$NASeq!="")
dataBnaive<-subset(dataBn,dataBn$RTIList=='None')
dataBtreated<-subset(dataBn,dataBn$RTIList!='None')

dataBnaive_unique<-dataBnaive[!duplicated(dataBnaive$PtID),]
dataBtreated_unique<-dataBtreated[!duplicated(dataBtreated$PtID),]

nrow(data);nrow(dataB);nrow(dataBn);nrow(dataBnaive);nrow(dataBnaive_unique);nrow(dataBtreated);nrow(dataBtreated_unique)

# remove sequences with bad character
dataBnaive_unique2<-dataBnaive_unique[-grep('~',dataBnaive_unique$NASeq),]
dataBtreated_unique2<-dataBtreated_unique[-grep('~',dataBtreated_unique$NASeq),]

# combine  identifier en sequence
##export naive sequences but still not aligned
dataBnaive_unique3<-cbind(as.character(dataBnaive_unique2$PtID),as.character(dataBnaive_unique2$NASeq))
write.table(as.data.frame(dataBnaive_unique3),file='subtypeB-rt_naive.csv',row.names=F,quote=F,sep=',')
##export treated sequences but still not aligned
dataBtreated_unique3<-cbind(as.character(dataBtreated_unique2$PtID),as.character(dataBtreated_unique2$NASeq))
write.table(as.data.frame(dataBtreated_unique3),file='subtypeB-rt_treated.csv',row.names=F,quote=F,sep=',')
```

#### generate a aligned fasta file from the sequences 
vi to add '>' 
tr ',' '\n' < subtypeB-rt_naive.csv > subtypeB-rt_naive.fasta
tr ',' '\n' < subtypeB-rt_treated.csv > subtypeB-rt_treated.fasta

** transfer to server ** 
scp *fasta cai:/home/ktheys0
grep '\.' subtypeB-rt_naive.fasta  |wc

** pairwise alignment against HXB2 ** 
SequenceTool RT subtypeB-rt_naive.fasta --exportKind GlobalAlignment --exportAlphabet Nucleotides --exportWithInsertions no  > subtypeB-rt_naive_aligned.fasta
SequenceTool RT subtypeB-rt_treated.fasta --exportKind GlobalAlignment --exportAlphabet Nucleotides --exportWithInsertions no  > subtypeB-rt_treated_aligned.fasta

** remove hxb2 sequence**
vi subtypeB-rt_naive_aligned.fasta
vi subtypeB-rt_treated_aligned.fasta

** Convert nucleotide sequences to amino acids  **
SequenceTool PR subtypeB-rt_naive_aligned.fasta --exportKind PositionTable  > subtypeB-rt_naive_mut.csv
SequenceTool PR subtypeB-rt_treated_aligned.fasta --exportKind PositionTable   > subtypeB-rt_treated_mut.csv

** download from server ** 
scp cai:/home/ktheys0/subtypeB-rt*fasta .
scp cai:/home/ktheys0/subtypeB-rt*csv .


## Step 2:  R analysis for calculating frequencies  

### At the amino acid level  

#### Protease frequencies AA 
```{r}
# NAIVE SEQUENCES
datamutnaive<-read.csv('subtypeB-pr_naive_mut.csv')
mutnaive<-c()
for(i in 2:ncol(datamutnaive))
{
ref<-datamutnaive[1,i]
compare<-datamutnaive[2:nrow(datamutnaive),i]
mutnaive<-c(mutnaive,sum(compare[compare!=""]==ref)/length(compare[compare!=""])*100)
}
plot(mutnaive,xaxt='n')
axis(1,at=seq(1,100,by=10),seq(1,100,by=10))

# TREATED SEQUENCES 
datamuttreated<-read.csv('subtypeB-pr_treated_mut.csv')
muttreat<-c()
for(i in 2:ncol(datamuttreated))
{
    ref<-datamuttreated[1,i]
    compare<-datamuttreated[2:nrow(datamuttreated),i]
    muttreat<-c(muttreat,sum(compare[compare!=""]==ref)/length(compare[compare!=""])*100)
}
plot(muttreat,xaxt='n')
axis(1,at=seq(1,100,by=10),seq(1,100,by=10))
points(mutnaive,col='green')


plot(muttreat,mutnaive)
```

#### Reverse transcriptase AA frequencies 
```{r,eval=F}
# NAIVE
datamutnaive<-read.csv('subtypeB-rt_naive_mut.csv')
mutnaive<-c()
for(i in 2:ncol(datamutnaive))
{
    ref<-datamutnaive[1,i]
    compare<-datamutnaive[2:nrow(datamutnaive),i]
    mutnaive<-c(mutnaive,sum(compare[compare!=""]==ref)/length(compare[compare!=""])*100)
}
plot(mutnaive,xaxt='n')
axis(1,at=seq(1,100,by=10),seq(1,100,by=10))

# TREATED
datamuttreated<-read.csv('subtypeB-pr_treated_mut.csv')
muttreat<-c()
for(i in 2:ncol(datamuttreated))
{
    ref<-datamuttreated[1,i]
    compare<-datamuttreated[2:nrow(datamuttreated),i]
    muttreat<-c(muttreat,sum(compare[compare!=""]==ref)/length(compare[compare!=""])*100)
}
plot(muttreat,xaxt='n')
axis(1,at=seq(1,100,by=10),seq(1,100,by=10))
points(mutnaive,col='green')

plot(muttreat,mutnaive)
```

### At the nucleotide level 

#### Protease frequencies nt frequencies  
```{r}
#READ IN FILES  
library(ape);library(seqinr);library(scales)
consensusB<-read.dna('consprrt.fasta',format = "fasta",as.character=TRUE)
pr_B_naive<-read.dna('subtypeB-pr_naive_aligned.fasta', format = "fasta",as.character=TRUE)
pr_B_treated<-read.dna('subtypeB-pr_treated_aligned.fasta', format = "fasta",as.character=TRUE)
nrow(pr_B_naive);nrow(pr_B_treated)
```

```{r}
#CALCULATE FREQUENCIES OF TRANSITIONS  
## for naive sequences 
patfasta<-pr_B_naive  #naive
i<-1
freqPatTs_threshold_pr_naive<-data.frame()

for (j in 1:297){#for each site in the sequence
    WT=	consensusB[j] #what is WT at site j?
    #print(WT)
    if(j %in% seq(1,295,by=3)) {# first position
        goodsequences<-which(paste(patfasta[,j+1],patfasta[,j+2]) == paste(consensusB[c(j+1)],consensusB[(j+2)]))
        if (WT=="c"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(2,296,by=3))) {# second position
        goodsequences<-which(paste(patfasta[,j-1],patfasta[,j+1]) == paste(consensusB[c(j-1)],consensusB[(j+1)]))
        if (WT=="c"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(3,297,by=3))) {# third position
        goodsequences<-which(paste(patfasta[,j-2],patfasta[,j-1]) == paste(consensusB[c(j-2)],consensusB[(j-1)]))
        if (WT=="c"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_pr_naive[i,j]=length(which(patfasta[goodsequences,j]=="a"))/length(goodsequences)
        }}
}

## for treated sequences 
patfasta<-pr_B_treated
i<-1
freqPatTs_threshold_pr_treated<-data.frame()

for (j in 1:297){#for each site in the sequence
    #print(j)
    WT=	consensusB[j] #what is WT at site j?
    
    if(j %in% seq(1,295,by=3)) {# first position
        goodsequences<-which(paste(patfasta[,j+1],patfasta[,j+2]) == paste(consensusB[c(j+1)],consensusB[(j+2)]))
        if (WT=="c"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(2,296,by=3))) {# second position
        goodsequences<-which(paste(patfasta[,j-1],patfasta[,j+1]) == paste(consensusB[c(j-1)],consensusB[(j+1)]))
        if (WT=="c"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(3,297,by=3))) {# third position
        goodsequences<-which(paste(patfasta[,j-2],patfasta[,j-1]) == paste(consensusB[c(j-2)],consensusB[(j-1)]))
        if (WT=="c"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_pr_treated[i,j]=length(which(patfasta[goodsequences,j]=="a"))/length(goodsequences)
        }}
}

## write 
write.table(as.data.frame(freqPatTs_threshold_pr_naive),file='subtypeB-frequencies_stanford_pr_naive.csv',row.names=F,quote=F,sep=',')
write.table(as.data.frame(freqPatTs_threshold_pr_treated),file='subtypeB-frequencies_stanford_pr_treated.csv',row.names=F,quote=F,sep=',')
```



#### Reverse transcriptase  frequencies nt frequencies  
```{r}
#READ IN FILES  
rt_B_naive<-read.dna('subtypeB-rt_naive_aligned.fasta', format = "fasta",as.character=TRUE)
rt_B_treated<-read.dna('subtypeB-rt_treated_aligned.fasta', format = "fasta",as.character=TRUE)
nrow(rt_B_naive);nrow(rt_B_treated)

#actually, an alignment against both PR and RT is the best for the numbering. For now only against RT so make a consensusB of only RT 
consensusB_rt<-consensusB[298:1977] 
```

```{r}
#CALCULATE FREQUENCIES OF TRANSITIONS  
## for naive sequences 
patfasta_rt<-rt_B_naive  #naive
i<-1
freqPatTs_threshold_rt_naive<-data.frame()

for (j in 1:1680){#for each site in the sequence
    #print(j)
    WT=	consensusB_rt[j] #what is WT at site j?
    
    if(j %in% seq(1,1678,by=3)) {# first position
        goodsequences<-which(paste(patfasta_rt[,j+1],patfasta_rt[,j+2]) == paste(consensusB_rt[c(j+1)],consensusB_rt[(j+2)]))
        
        if (WT=="c"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(2,1679,by=3))) {# second position
        goodsequences<-which(paste(patfasta_rt[,j-1],patfasta_rt[,j+1]) == paste(consensusB_rt[c(j-1)],consensusB_rt[(j+1)]))
        if (WT=="c"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(3,1680,by=3))) {# third position
        goodsequences<-which(paste(patfasta_rt[,j-2],patfasta_rt[,j-1]) == paste(consensusB_rt[c(j-2)],consensusB_rt[(j-1)]))
        if (WT=="c"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_rt_naive[i,j]=length(which(patfasta_rt[goodsequences,j]=="a"))/length(goodsequences)
        }}
}



## for treated sequences 
patfasta<-rt_B_treated  #treated
i<-1
freqPatTs_threshold_rt_treated<-data.frame()
for (j in 1:1680){#for each site in the sequence
    #print(j)
    WT=	consensusB_rt[j] #what is WT at site j?
    
    if(j %in% seq(1,1678,by=3)) {# first position
        goodsequences<-which(paste(patfasta_rt[,j+1],patfasta_rt[,j+2]) == paste(consensusB_rt[c(j+1)],consensusB_rt[(j+2)]))
        
        if (WT=="c"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(2,1679,by=3))) {# second position
        goodsequences<-which(paste(patfasta_rt[,j-1],patfasta_rt[,j+1]) == paste(consensusB_rt[c(j-1)],consensusB_rt[(j+1)]))
        if (WT=="c"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="a"))/length(goodsequences)
        }}
    
    if((j %in% seq(3,1680,by=3))) {# third position
        goodsequences<-which(paste(patfasta_rt[,j-2],patfasta_rt[,j-1]) == paste(consensusB_rt[c(j-2)],consensusB_rt[(j-1)]))
        if (WT=="c"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="t"))/length(goodsequences)}
        if (WT=="t"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="c"))/length(goodsequences)}
        if (WT=="a"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="g"))/length(goodsequences)}
        if (WT=="g"){freqPatTs_threshold_rt_treated[i,j]=length(which(patfasta_rt[goodsequences,j]=="a"))/length(goodsequences)
        }}
}

# export 
write.table(as.data.frame(freqPatTs_threshold_rt_naive),file='subtypeB-frequencies_stanford_rt_naive.csv',row.names=F,quote=F,sep=',')
write.table(as.data.frame(freqPatTs_threshold_rt_treated),file='subtypeB-frequencies_stanford_rt_treated.csv',row.names=F,quote=F,sep=',')
```


```{r}
# add PR and RT together. 
#1977
colnames(freqPatTs_threshold_pr_naive)<-1:297
colnames(freqPatTs_threshold_rt_naive)<-298:1977

#
freqPatTs_threshold_pr_rt_naive<-as.data.frame(c(freqPatTs_threshold_pr_naive,freqPatTs_threshold_rt_naive))
freqPatTs_threshold_pr_rt_treated<-as.data.frame(c(freqPatTs_threshold_pr_treated,freqPatTs_threshold_rt_treated))
```



## Step 3:  R analysis for assigning syn or nonsym and combine with frequencies 
```{r}
## Make a dataframe including information on syn vs nonsyn 

# Read in the csv file 

# intermezzo: read the correct files but for now: copy code
transition<-function(nuc){
    if (nuc=="a") return("g")
    if (nuc=="g") return("a")
    if (nuc=="c") return("t")
    if (nuc=="t") return("c")
}

typeofsitefunction<-function(WTcodon, mutantcodon){
    WTAA<-seqinr::translate(WTcodon)
    MUTAA<-seqinr::translate(mutantcodon)
    if (WTAA == MUTAA) return ("syn")
    else if (MUTAA == "*") return ("stop")
    else return ("nonsyn")
}

TypeOfSite<-c()
for (codon in 1:(length(consensusB)/3)){#for each codon in the sequence
    positions <- c(codon*3-2,codon*3-1, codon*3)
    WTcodon <- consensusB[positions]
    mutant1codon <- c(transition(WTcodon[1]), WTcodon[2:3])
    mutant2codon <- c(WTcodon[1],transition(WTcodon[2]), WTcodon[3])
    mutant3codon <- c(WTcodon[1:2], transition(WTcodon[3]))
    TypeOfSite<-c(TypeOfSite,typeofsitefunction(WTcodon,mutant1codon))
    TypeOfSite<-c(TypeOfSite,typeofsitefunction(WTcodon,mutant2codon))
    TypeOfSite<-c(TypeOfSite,typeofsitefunction(WTcodon,mutant3codon))
}

source("../RResistanceMutations.r")
TypeOfSite_backup<-TypeOfSite
TypeOfSite[sort(c((RTImuts$pos*3)-2,(RTImuts$pos*3)-1,(RTImuts$pos*3)))+297]<-"res" #PSP: This is only the RT resistance sites
```


```{r}
#Combine stanford frequencies with syn/non 
OverviewDF_stanford<-data.frame(num=1:1977,t(freqPatTs_threshold_pr_rt_naive),TypeOfSite)
OverviewDF_stanford_984<-OverviewDF_stanford[1:984,]
```


## Step 4:  R analysis for plotting frequencies: correlation with bacheler dataset 

** Get Bacheler frequencies previously calculated ** 
```{r} 
# Calculate the mean values 
bacheler_threshold <-read.csv('../../Output/freqPatTsInclDay0-threshold.csv') 
bacheler_threshold<-bacheler_threshold[,-1]
bacheler_threshold_mean<-apply(bacheler_threshold, 2 , function(x){mean(as.numeric(as.character(x)),na.rm=T)})

# combined
OverviewDF_stanford_984_bach<-cbind(OverviewDF_stanford_984,bacheler_threshold_mean)
names(OverviewDF_stanford_984_bach)[2]<-'stanford_threshold_mean'

# export 
write.table(as.data.frame(OverviewDF_stanford_984_bach),file='master-subtypeB-frequencies_stanford_bach_prrt_naive.csv',row.names=F,quote=F,sep=',')



# plots all 
plot(OverviewDF_stanford_984_bach$stanford_threshold_mean,OverviewDF_stanford_984_bach$bacheler_threshold_mean,main="Stanford vs Bacheler for all positions")
cor.test(OverviewDF_stanford_984_bach$stanford_threshold_mean,OverviewDF_stanford_984_bach$bacheler_threshold_mean)

# PROTEASE
PR<-OverviewDF_stanford_984_bach[14:297,] # Remove first 13 positions of protease
plot(PR$stanford_threshold_mean,PR$bacheler_threshold_mean,main="Stanford vs Bacheler for protease positions",ylab='Bacheler',xlab='Stanford')
cor.test(PR$stanford_threshold_mean,PR$bacheler_threshold_mean)

plot(PR$stanford_threshold_mean[PR$TypeOfSite=='syn'],PR$bacheler_threshold_mean[PR$TypeOfSite=='syn'],main="Stanford vs Bacheler for SYN protease positions",ylab='Bacheler',xlab='Stanford')
cor.test(PR$stanford_threshold_mean[PR$TypeOfSite=='syn'],PR$bacheler_threshold_mean[PR$TypeOfSite=='syn'])

plot(PR$stanford_threshold_mean[PR$TypeOfSite!='syn'],PR$bacheler_threshold_mean[PR$TypeOfSite!='syn'],main="Stanford vs Bacheler for NONSYN protease positions",ylab='Bacheler',xlab='Stanford')
cor.test(PR$stanford_threshold_mean[PR$TypeOfSite!='syn'],PR$bacheler_threshold_mean[PR$TypeOfSite!='syn'])

# REVERSE TRANSCRIPTASE
RT_pre<-OverviewDF_stanford_984_bach[298:984,]
RT<-RT_pre[RT_pre$TypeOfSite!='res',]
plot(RT$stanford_threshold_mean,RT$bacheler_threshold_mean,main="Stanford vs Bacheler for RT positions",ylab='Bacheler',xlab='Stanford')
cor.test(RT$stanford_threshold_mean,RT$bacheler_threshold_mean)

plot(RT$stanford_threshold_mean[RT$TypeOfSite=='syn'],RT$bacheler_threshold_mean[RT$TypeOfSite=='syn'],main="Stanford vs Bacheler for SYN RT positions",ylab='Bacheler',xlab='Stanford')
cor.test(RT$stanford_threshold_mean[RT$TypeOfSite=='syn'],RT$bacheler_threshold_mean[RT$TypeOfSite=='syn'])

plot(RT$stanford_threshold_mean[RT$TypeOfSite!='syn'],RT$bacheler_threshold_mean[RT$TypeOfSite!='syn'],main="Stanford vs Bacheler for NONSYN RT positions",ylab='Bacheler',xlab='Stanford')
cor.test(RT$stanford_threshold_mean[RT$TypeOfSite=='syn'],RT$bacheler_threshold_mean[RT$TypeOfSite=='syn'])

```


## Step 5: some exploratory analysis 
Why do some positions have really high frequency values? 


```{r}
OverviewDF_stanford_984_bach[OverviewDF_stanford_984_bach$stanford_threshold_mean > 0.6,]
OverviewDF_stanford_984_bach[OverviewDF_stanford_984_bach$stanford_threshold_mean > 0.6,'num']
# protease positions
((OverviewDF_stanford_984_bach[OverviewDF_stanford_984_bach$stanford_threshold_mean > 0.6,'num'])/3)[1:3]
# rt positions
((OverviewDF_stanford_984_bach[OverviewDF_stanford_984_bach$stanford_threshold_mean > 0.6,'num']) /3)[4:9]

```



```{r}
test<-OverviewDF_stanford_984_bach[OverviewDF_stanford_984_bach$stanford_threshold_mean < 0.6,]


```