#Script to analyse the frequency data and associate with features. 
Using the Bacheler et al data

* Read the csv files 
* Perform necessary calcuations
* Plot results (eventually new script)

```{r}
setwd("~/Documents/Git/bachelerProject/Rscripts")
source('./baseRscript.R')
library(scales)
library(plotrix)

```

Read the stored frequencies rather than calculating frequencies again
```{r}
#read the stored data
#We read a file with WT 0 as threshold, meaning no threshold)
read.table("../Output/freqPatTs_Bacheler.csv",sep=",",header=TRUE,row.names=1)->freqPatTs0
#calculate mean frequencies
MeanFreq<-apply(freqPatTs0, 2 , mean, na.rm=TRUE)
```

Calculate mean freq for each site and bootstrap / no longer done
```{r}
#Get conf intervals by bootsstrapping
#PSP Nov 11 2015: I don't think we need te bootstrapping, so I put it in an if (FALSE) statement.
#PSP June 2017: it looks like we may need it after all. 
if (FALSE){
    btmeansTs<-data.frame(row.names=names(freqPatTs0)[1:984]) #each row is a site, each column is a bootstrapped mean
    numbootstraps=100
    for (j in 1:984){# start with the first site
        print(j)
        for (k in 1:numbootstraps){ # first iteration
            btmeansTs[j,k]=mean(sample(freqPatTs0[,j],length(freqPatTs0[,j]),replace = TRUE),na.rm=TRUE)
            }
        }
    btmeansSorted<-t(apply(btmeansTs,1,sort))
    lowerConf<-btmeansSorted[,floor(0.025*numbootstraps)]
    upperConf<-btmeansSorted[,floor((1-0.025)*numbootstraps)]
    }
```

## Create overview dataframe and plot site frequency spectra 
Only synonymous, non-synomous and stop codons are considered
- for each mutation, determine whether it is synonymous, non-synonymous or creates a stop
- add information on resistance  positions

```{r}
#PSP Nov 11 2015 I removed lowerConf and upperConf here because we no longer calculate them
#PSP June 2017 added them again (lowerConf and upperConf)
#PSP Nov 11 2015 I renamed x OverviewDF and newdata OverviewDFOrderedByFreq
OverviewDF<-data.frame(num=1:984,MeanFreq,TypeOfSite,lowerConf,upperConf)

#Mut rates and sel coefficients
read.csv("../Data/HIVMutRates/HIVMutRates.csv")->mutrates
OverviewDF$TSmutrate<-0
OverviewDF$TSmutrate[OverviewDF$WTnt=="a"]<-mutrates$Probability[mutrates$Nucleotide.substitution=="AG"]
OverviewDF$TSmutrate[OverviewDF$WTnt=="c"]<-mutrates$Probability[mutrates$Nucleotide.substitution=="CU"]
OverviewDF$TSmutrate[OverviewDF$WTnt=="g"]<-mutrates$Probability[mutrates$Nucleotide.substitution=="GA"]
OverviewDF$TSmutrate[OverviewDF$WTnt=="t"]<-mutrates$Probability[mutrates$Nucleotide.substitution=="UC"]

for (i in 1:984){
    OverviewDF$EstSelCoeff[i] = EstimatedS(OverviewDF$TSmutrate[i],OverviewDF$MeanFreq[i])
    }

OverviewDF$WTnt<-consensusB
#WT AAs 
OverviewDF$WTAA<-""
for (i in 1:984){
    if (i%%3==1) OverviewDF$WTAA[i] = seqinr::translate(OverviewDF$WTnt[c(i,i+1,i+2)])
    if (i%%3==2) OverviewDF$WTAA[i] = seqinr::translate(OverviewDF$WTnt[c(i-1,i,i+1)])
    if (i%%3==0) OverviewDF$WTAA[i] = seqinr::translate(OverviewDF$WTnt[c(i-2,i-1,i)])
}

OverviewDF$MUTAA<-""
#MUT AAs
for (i in 1:984){
    if (i%%3==1) OverviewDF$MUTAA[i] = seqinr::translate(c(transition(OverviewDF$WTnt[i]),OverviewDF$WTnt[c(i+1,i+2)]))
    if (i%%3==2) OverviewDF$MUTAA[i] = seqinr::translate(c(OverviewDF$WTnt[c(i-1)],transition(OverviewDF$WTnt[i]),OverviewDF$WTnt[c(i+1)]))
    if (i%%3==0) OverviewDF$MUTAA[i] = seqinr::translate(c(OverviewDF$WTnt[c(i-2,i-1)],transition(OverviewDF$WTnt[i])))
}

#Add whether AA change is drastic 
OverviewDF$bigAAChange<-0

for(i in 1:nrow(OverviewDF)){
    WT <- amCat(OverviewDF[i,'WTAA'])
    MUT <- amCat(OverviewDF[i,'MUTAA'])
    if (WT == MUT){ OverviewDF$bigAAChange[i] <- 0 
    }else{
        OverviewDF$bigAAChange[i] <- 1
    }
}

#Add whetehr makes CpG 
OverviewDF$makesCpG <- 0
for(i in 1:nrow(OverviewDF)){
    trip <- OverviewDF$WTnt[c(i-1, i, i + 1)]
    if(trip[1] == "c" & trip[2] == "a" ){
        OverviewDF$makesCpG[i] <- 1
    }
    if(trip[2] == "t" & trip[3] == "g"){
       OverviewDF$makesCpG[i] <- 1
    }
}

OverviewDF$ProRT<-c(rep("Pro",297),rep("RT",687))
OverviewDF$AAnum<-c(sort(rep(1:99,3)),sort(rep(1:229,3)))
write.csv(OverviewDF,"../Output/OverviewSelCoeff_Bacheler.csv")

#colors
OverviewDFOrderedByFreq <- OverviewDF[order(MeanFreq),] 
OverviewDFOrderedByFreq$color<-""
for (i in 1:984){
    if (OverviewDFOrderedByFreq$TypeOfSite[i]=="syn") OverviewDFOrderedByFreq$color[i] = "darkolivegreen3"
    if (OverviewDFOrderedByFreq$TypeOfSite[i]=="nonsyn") OverviewDFOrderedByFreq$color[i] = "red"
    if (OverviewDFOrderedByFreq$TypeOfSite[i]=="stop") OverviewDFOrderedByFreq$color[i] = "black"
    if (OverviewDFOrderedByFreq$TypeOfSite[i]=="res") OverviewDFOrderedByFreq$color[i] = "purple"
}
```

Test whether non syn muts, syn muts and nonsense muts are different in freq
```{r}
FreqsSyn<-OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="syn"]
FreqsNonSyn<-OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="nonsyn"]
FreqsStop<-OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="stop"]

wilcox.test(FreqsSyn, FreqsNonSyn,alternative = "greater", paired = FALSE)
wilcox.test(FreqsNonSyn,FreqsStop,alternative = "greater", paired = FALSE)

```


Make a figure with the selection coefficients across Protease
```{r}
pdf("../Output/EstSelCoeffPRO.pdf",width=12,height=8)
par(mfrow=c(1,1))
#make log scale
plot(OverviewDF$num[40:297],OverviewDF$EstSelCoeff[40:297],log="y", 
     ylab="Estimated Selection Coefficient (cost)", xlab = "Position in Protease", xaxt="n",yaxt="n",col="darkgrey",t="c",pch=".",
     ylim=c(0.9*10^-5,1))
axis(1,at=3*c(14,35,55,75,95)-1,labels=c(14,35,55,75,95))
axis(2,at=c(10^-5,10^-4,10^-3,10^-2,10^-1,10^-0),labels=c(10^-5,10^-4,10^-3,10^-2,10^-1,10^-0))
#Add colors for different types of sites

#A #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num<=297],pch=21,bg=brewer.pal(11, "Spectral")[11])

points(45,2*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[11])
text(48,2*10^-5,pos=4,"A/T/C, syn, no CpG")

#T #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num<=297],pch=21,bg=brewer.pal(11, "Spectral")[11])

#C #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num<=297],pch=21,bg=brewer.pal(11, "Spectral")[11])

#A #SYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num<=297],pch=24,bg=brewer.pal(11, "Spectral")[10])

points(45,1.4*10^-5,pch=24,bg=brewer.pal(11, "Spectral")[10])
text(48,1.4*10^-5,pos=4,"A/T, syn, CpG")

#T #SYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num<=297],pch=24,bg=brewer.pal(11, "Spectral")[10])

#G #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num<=297],pch=25,bg=brewer.pal(11, "Spectral")[8],col=brewer.pal(11, "RdYlGn")[11])

points(45,1*10^-5,pch=25,bg=brewer.pal(11, "Spectral")[8],col=brewer.pal(11, "RdYlGn")[11])
text(48,1*10^-5,pos=4,"G, syn")

#A #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

points(100,2*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[7],col=2)
text(103,2*10^-5,pos=4,"A/T, non-syn, no drastic AA change")

#T #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

#A #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

#T #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

#With big AA change

#A #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])

points(100,1.4*10^-5,pch=22,bg=brewer.pal(11, "Spectral")[5],col=2)
text(103,1.4*10^-5,pos=4,"A/T, non-syn, drastic AA change")

#T #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])

#A #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])

#T #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num<=297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])


#no big AA change

#C #NONSYN no big AA change
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[3])

points(100,1*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[3],col=2)
text(103,1*10^-5,pos=4,"C, non-syn, no drastic AA change")

#G #NONSYN no big AA change
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[1])

points(200,2*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[1],col=2)
text(203,2*10^-5,pos=4,"G, non-syn, no drastic AA change")

#C #NONSYN bigAAChange
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[3])

points(200,1.4*10^-5,pch=22,bg=brewer.pal(11, "Spectral")[3],col=2)
text(203,1.4*10^-5,pos=4,"C, non-syn, drastic AA change")

#G #NONSYN bigAAChange
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num<=297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[1])

points(200,1*10^-5,pch=22,bg=brewer.pal(11, "Spectral")[1],col=2)
text(203,1*10^-5,pos=4,"G, non-syn, drastic AA change")

#nonsense
points(OverviewDF$num[OverviewDF$TypeOfSite=="stop"&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="stop"&OverviewDF$num<=297],pch=22,col=2,bg=1)

points(280,2*10^-5,pch=22,bg=1,col=1)
text(283,2*10^-5,pos=4,"nonsense")

#resistance
points(OverviewDF$num[OverviewDF$TypeOfSite=="res"&OverviewDF$num<=297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="res"&OverviewDF$num<=297],pch=8,col=1,bg=1)

points(280,1.4*10^-5,pch=8,bg=1,col=1)
text(283,1.4*10^-5,pos=4,"resistance")

dev.off()
```

Make a figure with the selection coefficients across RT
```{r}
pdf("../Output/EstSelCoeffRT.pdf",width=12,height=8)
par(mfrow=c(1,1))
#make log scale
plot(OverviewDF$num[298:984],OverviewDF$EstSelCoeff[298:984],log="y", 
     ylab="Estimated Selection Coefficient (cost)", xlab = "Position in Reverse Transcriptase", xaxt="n",yaxt="n",col="darkgrey",
     t="c",pch=".",
     ylim=c(0.9*10^-5,1))
axis(1,at=298+3*seq(20,228,by=20)-1,labels=seq(20,228,by=20))
axis(2,at=c(10^-5,10^-4,10^-3,10^-2,10^-1,10^-0),labels=c(10^-5,10^-4,10^-3,10^-2,10^-1,10^-0))
#Add colors for different types of sites

#A #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num>297],pch=21,bg=brewer.pal(11, "Spectral")[11])

points(298+45,2*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[11])
text(298+48,2*10^-5,pos=4,"A/T/C, syn, no CpG")

#T #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num>297],pch=21,bg=brewer.pal(11, "Spectral")[11])

#C #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num>297],pch=21,bg=brewer.pal(11, "Spectral")[11])

#A #SYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num>297],pch=24,bg=brewer.pal(11, "Spectral")[10])

points(298+45,1.4*10^-5,pch=24,bg=brewer.pal(11, "Spectral")[10])
text(298+48,1.4*10^-5,pos=4,"A/T, syn, CpG")

#T #SYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num>297],pch=24,bg=brewer.pal(11, "Spectral")[10])

#G #SYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="syn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num>297],pch=25,bg=brewer.pal(11, "Spectral")[8],col=brewer.pal(11, "RdYlGn")[11])

points(298+45,1*10^-5,pch=25,bg=brewer.pal(11, "Spectral")[8],col=brewer.pal(11, "RdYlGn")[11])
text(298+48,1*10^-5,pos=4,"G, syn")

#A #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

points(298+180,2*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[7],col=2)
text(298+183,2*10^-5,pos=4,"A/T, non-syn, no drastic AA change")

#T #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

#A #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

#T #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[7])

#With big AA change

#A #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])

points(298+180,1.4*10^-5,pch=22,bg=brewer.pal(11, "Spectral")[5],col=2)
text(298+183,1.4*10^-5,pos=4,"A/T, non-syn, drastic AA change")

#T #NONSYN #no CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])

#A #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="a"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])

#T #NONSYN #CPG
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="t"&OverviewDF$makesCpG==1&OverviewDF$num>297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[5])


#no big AA change

#C #NONSYN no big AA change
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[3])

points(298+180,1*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[3],col=2)
text(298+183,1*10^-5,pos=4,"C, non-syn, no drastic AA change")

#G #NONSYN no big AA change
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==0],pch=21,col=2,bg=brewer.pal(11, "Spectral")[1])

points(298+400,2*10^-5,pch=21,bg=brewer.pal(11, "Spectral")[1],col=2)
text(298+403,2*10^-5,pos=4,"G, non-syn, no drastic AA change")

#C #NONSYN bigAAChange
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="c"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[3])

points(298+400,1.4*10^-5,pch=22,bg=brewer.pal(11, "Spectral")[3],col=2)
text(298+403,1.4*10^-5,pos=4,"C, non-syn, drastic AA change")

#G #NONSYN bigAAChange
points(OverviewDF$num[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="nonsyn"&OverviewDF$WTnt=="g"&OverviewDF$makesCpG==0&OverviewDF$num>297&OverviewDF$bigAAChange==1],pch=22,col=2,bg=brewer.pal(11, "Spectral")[1])

points(298+400,1*10^-5,pch=22,bg=brewer.pal(11, "Spectral")[1],col=2)
text(298+403,1*10^-5,pos=4,"G, non-syn, drastic AA change")

#nonsense
points(OverviewDF$num[OverviewDF$TypeOfSite=="stop"&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="stop"&OverviewDF$num>297],pch=22,col=2,bg=1)

points(298+620,2*10^-5,pch=22,bg=1,col=1)
text(298+623,2*10^-5,pos=4,"nonsense")

#resistance
points(OverviewDF$num[OverviewDF$TypeOfSite=="res"&OverviewDF$num>297],OverviewDF$EstSelCoeff[OverviewDF$TypeOfSite=="res"&OverviewDF$num>297],pch=8,col=1,bg=1)

points(298+620,1.4*10^-5,pch=8,bg=1,col=1)
text(298+623,1.4*10^-5,pos=4,"resistance")

dev.off()
```

Make a figure with all single site frequency spectra for Protease (and then RT)
```{r}
pdf("../Output/SingleSiteFrequencySpectraPRO.pdf",width=8,height=10)
par(mfrow=c(3,3))
for (i in 1:297){
    #Site excluded bc too many non-consensus patients? Threshold = 10%
    excluded = ""; if (NumPats66Excluded[i]>0.1) excluded = "(site excluded)"
    resistancesite = ""; if (OverviewDFOrderedByFreq$TypeOfSite[which(OverviewDFOrderedByFreq$num==i)]=="res") resistancesite = "\n resistance codon, excluded"
    #first create empty plot with title
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",col=0,border=0,main = paste("Protease site", i,"\n","AA",ceiling(i/3),excluded,resistancesite),xlab="Frequency",ylab="Count")
    #Next, show true height of 0 bar
    if (length(which(freqPatTs0[,i]<0.02))>=70){
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",col=OverviewDFOrderedByFreq$color[which(OverviewDFOrderedByFreq$num==i)],add=T)}
    #next show all data (unfiltered), but only until 50 for 0 cat
    hist(c(rep(0,min(60,length(which(freqPatTs0[,i]<0.02)))),freqPatTs0[,i][which(freqPatTs0[,i]>0)]),breaks=seq(0,1,by=0.02),add=T,
         col=OverviewDFOrderedByFreq$color[which(OverviewDFOrderedByFreq$num==i)])
    axis(2,labels = c(10,20,30,40,50,max(70,length(which(freqPatTs0[,i]<0.02)))), at = c(10,20,30,40,50,70), las=1)
    if (length(which(freqPatTs0[,i]<0.02))>=70){
    axis.break(axis=2,breakpos=60,bgcol="white",breakcol="black",style="slash",brw=0.02)
    points(c(0.01,0.02),c(60,60),pch=15,cex=2.5,col="white")
    }else{axis(2,labels = 60,at=60,las=1)}
}
dev.off()
```

Make a figure with single site frequency spectra for Protease AA 58
```{r}
 pdf("../Output/0.0030585683",width=8,height=4)
cols <- c(0,brewer.pal(6, "Set2")[c(2, 1)])
par(mfrow=c(1,3))
for (i in 172:174){
    #first create empty plot with title
if (i == 172){
    t=paste("C172T \n nonsense mutation",sep="")
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",
    col=cols[1],border=0,
#    main= bquote(paste(.(t),(C %->% T ))), cex=1.3,
    main="",cex=1.2,
    xlab="Frequency", ylab="Count",cex.lab=1.4)
    title(t,cex=1.2,line=0)
    }
if (i == 173){
    t=paste("A173G \n non-synonymous mutation",sep="")
#    t=paste("Protease: site ", i,"\n non-synonymous mutation",sep="")
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",
    col=cols[2],border=0,
#    main = bquote(paste(.(t),(A %->% G ))), cex=1.3,
    main= "", cex=1.2,
    xlab="Frequency", ylab="Count",cex.lab=1.4)
    title(t,cex=1.2,line=0)
    }
if (i == 174){
    t=paste("G174A \n synonymous mutation",sep="")
#    t=paste("Protease: site ", i,"\n synonymous mutation",sep="")
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",
    col=cols[3],border=0,
#    main = bquote(paste(.(t),(G %->% A ))), cex=1.3,
    main= "", cex=1.2,
    xlab="Frequency", ylab="Count",cex.lab=1.4)
    title(t,cex=1.2,line=0)
    }
    #Next, show true height of 0 bar
if (i == 172){
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),
         yaxt="n",col=OverviewDFOrderedByFreq$color[which(OverviewDFOrderedByFreq$num==i)],add=T)}
if (i == 173){
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),
         yaxt="n",col=cols[2],add=T)}
if (i == 174){
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),
         yaxt="n",col=cols[3],add=T)}
    #next show all data (unfiltered), but only until 50 for 0 cat

if (i == 172){
    hist(c(rep(0,min(60,length(which(freqPatTs0[,i]<0.02)))),freqPatTs0[,i][which(freqPatTs0[,i]>0)]),
         breaks=seq(0,1,by=0.02),add=T,
         col=OverviewDFOrderedByFreq$color[which(OverviewDFOrderedByFreq$num==i)])}
if (i == 173){
    hist(c(rep(0,min(60,length(which(freqPatTs0[,i]<0.02)))),freqPatTs0[,i][which(freqPatTs0[,i]>0)]),
         breaks=seq(0,1,by=0.02),add=T,
         col=cols[2])}
if (i == 174){
    hist(c(rep(0,min(60,length(which(freqPatTs0[,i]<0.02)))),freqPatTs0[,i][which(freqPatTs0[,i]>0)]),
         breaks=seq(0,1,by=0.02),add=T,
         col=cols[3])}

    axis(2,labels = c(10,20,30,40,50,max(70,length(which(freqPatTs0[,i]<0.02)))), 
         at = c(10,20,30,40,50,70), las=1)
    if (length(which(freqPatTs0[,i]<0.02))>=70){
    axis.break(axis=2,breakpos=60,bgcol="white",breakcol="black",style="slash",brw=0.02)
    points(c(0.01,0.02),c(60,60),pch=15,cex=2.5,col="white")
    }else{axis(2,labels = 60,at=60,las=1)}
}
dev.off()
```

Make a figure with all single site frequency spectra for RT
```{r}
pdf("../Output/SingleSiteFrequencySpectraRT.pdf",width=8,height=10)
par(mfrow=c(3,3))
for (i in 298:length(OverviewDFOrderedByFreq$color)){
    #Site excluded bc too many non-consensus patients? Threshold = 10%
    excluded = ""; if (NumPats66Excluded[i]>0.1) excluded = "(site excluded)"
    resistancesite = ""; if (OverviewDFOrderedByFreq$TypeOfSite[which(OverviewDFOrderedByFreq$num==i)]=="res") resistancesite = "\n resistance codon, excluded"
    #first create empty plot with title
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",col=0,border=0,main = paste("RT site", i-297,"\n","AA",ceiling((i-297)/3),excluded,resistancesite),xlab="Frequency",ylab="Count")
    #Next, show true height of 0 bar
    if (length(which(freqPatTs0[,i]<0.02))>=70){
    hist(rep(0,70),breaks=seq(0,1,by=0.02),xlim=c(0,1),ylim=c(0,70),yaxt="n",col=OverviewDFOrderedByFreq$color[which(OverviewDFOrderedByFreq$num==i)],add=T)}
    #next show all data (unfiltered), but only until 50 for 0 cat
    hist(c(rep(0,min(60,length(which(freqPatTs0[,i]<0.02)))),freqPatTs0[,i][which(freqPatTs0[,i]>0)]),breaks=seq(0,1,by=0.02),add=T,
         col=OverviewDFOrderedByFreq$color[which(OverviewDFOrderedByFreq$num==i)])
    axis(2,labels = c(10,20,30,40,50,max(70,length(which(freqPatTs0[,i]<0.02)))), at = c(10,20,30,40,50,70), las=1)
    if (length(which(freqPatTs0[,i]<0.02))>=70){
    axis.break(axis=2,breakpos=60,bgcol="white",breakcol="black",style="slash",brw=0.02)
    points(c(0.01,0.02),c(60,60),pch=15,cex=2.5,col="white")
    }else{axis(2,labels = 60,at=60,las=1)}
}
dev.off()
```

PRO: Make the plots (transitions) for ranking
```{r}
pdf("../Output/ProteaseRanking.pdf",width = 13, height = 10)
PROdata<-OverviewDFOrderedByFreq[OverviewDFOrderedByFreq$num<298,]
#remove resistance mutations
PROdata<-PROdata[PROdata$TypeOfSite!="res",]
#remove positions with too many non-consensus patients
which(NumPats66Excluded>0.1)->listSitesToExclude
PROdata<-PROdata[-which(PROdata$num %in% listSitesToExclude),]
plot(log(PROdata$MeanFreq+0.001), main = "Protease mutant frequencies",
     ylim=c(log(0.001),log(0.5)),cex=1.5, pch = 16, col=alpha(PROdata$color, 1), xlab = "Nucleotides ordered by mean mutation frequency", ylab = "Mean mutation frequency" , yaxt = "n")
axis(2,labels = c(0,0.001, 0.005, 0.05, 0.1), at = log(c(0.001, 0.002, 0.006, 0.051, 0.101)),las=1)
points(10*PROdata$NumPats66Excluded+log(0.001),pch=16,cex=0.5,col="grey")
axis(4,labels = c(0,0.05,0.1), at = 10*c(0,0.05,0.1)+log(0.001),las=1)
dev.off()
```

Show the effect of our filtering out non-consensus patients at day 0
```{r}
pdf("../Output/ProteaseRanking_EffectFiltering.pdf",width = 13, height = 10)
plot(log(PROdata$MeanFreq+0.001), main = "Protease mutant frequencies",
     ylim=c(log(0.001),log(0.5)),cex=1.5, pch = 16, col=alpha(PROdata$color, 1), xlab = "Nucleotides ordered by mean mutation frequency", ylab = "Mean mutation frequency" , yaxt = "n")
axis(2,labels = c(0,0.001, 0.005, 0.05, 0.1), at = log(c(0.001, 0.002, 0.006, 0.051, 0.101)),las=1)
#points(10*PROdata$NumPats66Excluded+log(0.001),pch=16,cex=0.5,col="grey")
#axis(4,labels = c(0,0.05,0.1), at = 10*c(0,0.05,0.1)+log(0.001),las=1)
points(log(PROdata$colMeansTs66[order(PROdata$colMeansTs66)]+0.001),cex=1.5, pch = 16, col=alpha(PROdata$color, 0.3))
dev.off()
```

RT: Make the plots (transitions) for rankng
```{r}
#pdf("../Output/RTRanking.pdf",width = 13, height = 10)
RTdata<-OverviewDFOrderedByFreq[OverviewDFOrderedByFreq$num>=298,]
RTdata<-RTdata[RTdata$TypeOfSite!="res",]
#remove positions with too many non-consensus patients
which(NumPats66Excluded>0.1)->listSitesToExclude
RTdata<RTdata[-which(RTdata$num %in% listSitesToExclude),]
plot(log(RTdata$MeanFreq+0.001), main = "RT mutant frequencies",
     ylim=c(log(0.001),log(0.5)),cex=1.5, pch = 16, col=alpha(RTdata$color, 1), xlab = "Nucleotides ordered by mean mutation 
frequency", ylab = "Mean mutation frequency" , yaxt = "n")
axis(2,labels = c(0,0.001, 0.005, 0.05, 0.1), at = log(c(0.001, 0.002, 0.006, 0.051, 0.101)),las=1)
points(10*PROdata$NumPats66Excluded+log(0.001),pch=16,cex=0.5,col="grey")
axis(4,labels = c(0,0.05,0.1), at = 10*c(0,0.05,0.1)+log(0.001),las=1)
#dev.off()
```

Show the effect of our filtering out non-consensus patients at day 0
```{r}
pdf("../Output/RTRanking_EffectFiltering.pdf",width = 13, height = 10)
plot(log(RTdata$MeanFreq+0.001), main = "RT mutant frequencies",
     ylim=c(log(0.001),log(0.5)),cex=1.5, pch = 16, col=alpha(PROdata$color, 1), xlab = "Nucleotides ordered by mean mutation frequency", ylab = "Mean mutation frequency" , yaxt = "n")
axis(2,labels = c(0,0.001, 0.005, 0.05, 0.1), at = log(c(0.001, 0.002, 0.006, 0.051, 0.101)),las=1)
#points(10*PROdata$NumPats66Excluded+log(0.001),pch=16,cex=0.5,col="grey")
#axis(4,labels = c(0,0.05,0.1), at = 10*c(0,0.05,0.1)+log(0.001),las=1)
points(log(RTdata$colMeansTs66[order(RTdata$colMeansTs66)]+0.001),cex=1.5, pch = 16, col=alpha(RTdata$color, 0.3))
dev.off()
```

Show distribution of fitness effects estimated
```{r}
par(mfrow=c(2,2))
hist(log10(1.11*10^-5/OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="nonsyn"&consensusB=="a"]),breaks= 20, xlim=c(-5,-0),ylim=c(0,70),col="red",main="A",xlab="sel coeff (10log)",ylab="count")
hist(log10(2.41*10^-5/OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="nonsyn"&consensusB=="c"]),breaks= 15, xlim=c(-5,-0),ylim=c(0,20),col="red",main="C",xlab="sel coeff (10log)",ylab="count")
hist(log10(5.48*10^-5/OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="nonsyn"&consensusB=="g"]),breaks= 15, xlim=c(-5,-0),ylim=c(0,30),col="red",main="G",xlab="sel coeff (10log)",ylab="count")
hist(log10(1.11*10^-5/OverviewDF$MeanFreq[OverviewDF$TypeOfSite=="nonsyn"&consensusB=="t"]),breaks= 15,xlim=c(-5,-0),ylim=c(0,30),col="red",main="T",xlab="sel coeff (10log)",ylab="count")
```

Make an AA variability map 
```{r}
translatedB<-seqinr::translate(consensusB)
#convert the fasta files into AA sequences
translatedfasta<-c()
for (i in 1:length(listfastafiles)){ #for each fastafile 
         filename=paste("../Data/BachelerFiles/FASTAfiles/",substr(listfastafiles[i],1,6),".fasta",sep="")
	       patfasta<-read.dna(filename, format = "fasta",as.character=TRUE) #read the file       
	       translatedfasta<-rbind(translatedfasta,t(apply(patfasta,1,function(x) seqinr::translate(x))))
}
```

Determine for each codon position, the frequency of mutations compared to subtype B
```{r}
mutations<-c()
for(m in 1:length(translatedB)){
prevmut<-100-(length(which(translatedfasta[,m]==translatedB[m]))/nrow(translatedfasta)*100)
mutations<-c(mutations,prevmut)
}

# Plot for protease including resistance positions
plot(mutations[1:99],type='o',ylab='Mutation frequency (%)',xlab='AA position')
# Plot for RT including resistance positions
plot(mutations[ 100:328],type='o',ylab='Mutation frequency (%)',xlab='AA position',xaxt='n')
axis(1,(100:328)-99,label=(100:328)-99)


# Plot for RT excluding resistance positions and including resisatnce positions in red
mutationposition<-cbind(1:229,mutations[100:328])
plot(mutationposition[-c(sort(RTImuts$pos)),1],mutationposition[-c(sort(RTImuts$pos)),2],type='o',ylab='Mutation frequency (%)',xlab='AA position',xaxt='n')
points(mutationposition[c(sort(RTImuts$pos)),1],mutationposition[c(sort(RTImuts$pos)),2],col='red')
axis(1,(100:328)-99,label=(100:328)-99)
```

Non-Syn plot 
```{r}
pdf("../Output/PR-nonsyn.pdf",width = 13, height = 10)
PROdataNonSyn<-PROdata[PROdata$TypeOfSite=="nonsyn",]
#Make plot of Protease 
plot(PROdataNonSyn$num[order(PROdataNonSyn$num)],PROdataNonSyn$colMeansTs66[order(PROdataNonSyn$num)]+0.001,col=PROdataNonSyn$color[order(PROdataNonSyn$num)],pch=16,log="y",type="b", xlab = "Position in protease gene", ylab = "frequency non-synonymous transitions")#,ylim=c(0,max(PROdataNonSyn$colMeansTs66)))
#Add moving mean
for (i in 20:280){
    range = (i-20):(i+20)
#    points(i, mean(PROdataNonSyn$colMeansTs66[PROdataNonSyn$num%in%range]), col="blue",pch = 12, cex=3)
    points(i, median(PROdataNonSyn$colMeansTs66[PROdataNonSyn$num%in%range]), col="green",pch = 11, cex=2)
}
dev.off()

RTdataNonSyn<-RTdata[RTdata$TypeOfSite=="nonsyn",]
plot(RTdataNonSyn$num[order(RTdataNonSyn$num)],RTdataNonSyn$colMeansTs66[order(RTdataNonSyn$num)]+0.001,col=RTdataNonSyn$color[order(RTdataNonSyn$num)],pch=16,log="y",type="b")

#Add moving mean
for (i in 370:900){
    range = (i-20):(i+20)
   # points(i, mean(RTdataNonSyn$colMeansTs66[RTdataNonSyn$num%in%range]), col="blue",pch = 12, cex=3)
    points(i, median(RTdataNonSyn$colMeansTs66[RTdataNonSyn$num%in%range]), col="green",pch = 11, cex=3)
}

i=370
    range = (i-20):(i+20)
```


Syn plot 
```{r}
pdf("../Output/PR-syn.pdf",width = 13, height = 10)
PROdataSyn<-PROdata[PROdata$TypeOfSite=="syn",]
#Make plot of Protease 
plot(PROdataSyn$num[order(PROdataSyn$num)],PROdataSyn$colMeansTs66[order(PROdataSyn$num)]+0.001,col=PROdataSyn$color[order(PROdataSyn$num)],pch=16,
     log="y",
     type="b")#,ylim=c(0,max(PROdataNonSyn$colMeansTs66)))

#Add moving mean
for (i in 20:280){
    range = (i-10):(i+10)
    points(i, median(PROdataSyn$colMeansTs66[PROdataSyn$num%in%range]), col="blue",pch = 11, cex=2)
}
dev.off()
```


# Next steps 
Going to randomisations
- frequency-ordered figure: does the coloring scheme hold?
  
- position-ordered figure: does the pattern hold?  
    
###For protease, which are the most deleterious non-syn sites? 
To be done 
Interesting? why? for therapy or vaccine. 

###For protease, which are the most neutral non-syn sites? 
corresponding to the expected frequency based on sequence length * mutation rate? Frequency should flucatuate aroudn the expected value? 

#For protease, which are the most deleterious syn sites? 
To be done : green colored sites with low frequ
Interesting why?  HIV replication?


